web/src/beta/components/fields/common/NumberInput/index.tsx

```diff
@@ -0,0 +1,168 @@
+  suffix,
+  onChange,
+  min,
+  max,
+  disabled = false,
+}) => {
+  const [innerValue, setInnerValue] = useState<number | undefined>(value);
+  const [, setNotification] = useNotification();
+
+  const isEditing = useRef(false);
+  const t = useT();
+
+  const theme = useTheme();
+
+  useEffect(() => {
+    setInnerValue(value);
+  }, [value]);
+
+  const callChange = useCallback(
+    (newValue: number | undefined) => {
+      if (!onChange || !isEditing.current) {
+        return;
+      }
+
+      if (newValue === undefined) {
+        setInnerValue(undefined);
+        onChange(undefined);
+      } else if (typeof max === "number" && isFinite(max) && newValue > max) {
+        setNotification({ type: "warning", text: t("You have passed the maximum value.") });
+        setInnerValue(undefined);
+        onChange(undefined);
+      } else if (typeof min === "number" && isFinite(min) && newValue < min) {
+        setNotification({ type: "warning", text: t("You have passed the minimum value.") });
+        setInnerValue(undefined);
+        onChange(undefined);
+      } else if (!isNaN(newValue)) {
+        setInnerValue(newValue);
+        onChange(newValue);
+      }
+    },
+    [onChange, max, min, setNotification, t],
+  );
+
+  const handleChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = parseFloat(e.currentTarget.value);
+    setInnerValue(newValue);
+    const minWidth = Math.max(metricsSizes.xs, `${newValue}`.length * 10);
+    e.currentTarget.style.width = `${minWidth}px`;
+  }, []);
+
+  const handleKeyPress = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      if (e.key === "Enter") {
+        callChange(parseFloat(e.currentTarget.value));
+      }
+    },
+    [callChange],
+  );
+
+  const handleFocus = useCallback(() => {
+    isEditing.current = true;
+  }, []);
+
+  const handleBlur = useCallback(
+    (e: React.SyntheticEvent<HTMLInputElement>) => {
+      callChange(parseFloat(e.currentTarget.value));
+      isEditing.current = false;
+    },
+    [callChange],
+  );
+
+  return (
+    <Wrapper>
+      <FormWrapper inactive={!!disabled} className={className}>
+        <StyledInput
+          type="number"
+          value={innerValue}
+          disabled={disabled}
+          onChange={handleChange}
+          onKeyPress={handleKeyPress}
+          onFocus={handleFocus}
+          onBlur={handleBlur}
+          min={min}
+          max={max}
+          step="any"
+        />
+        {suffix && (
+          <Text size="footnote" color={theme.content.weak} otherProperties={{ userSelect: "none" }}>
+            {suffix}
+          </Text>
+        )}
+      </FormWrapper>
+      {inputDescription && (
+        <Text size="footnote" color={theme.content.weak}>
+          {inputDescription}
+        </Text>
+      )}
+    </Wrapper>
+  );
+};
+
+const Wrapper = styled.div`
+  width: 100%;
+  text-align: center;
+`;
+
+type FormProps = { inactive: boolean };
```

Comment:
Can you just move this directly in to the styled component to clean things up a bit

Reply:
done